Clean/Utils/Tactics/SplitProvableStructEq.lean

```diff
@@ -0,0 +1,155 @@
+          -- variable = struct_literal
+          match lhs with
+          | .fvar fvarId =>
+            -- Check if the variable has ProvableStruct
+            let varType ← inferType lhs
+            if ← hasProvableStructInstance varType then
+              structVarsToCase := fvarId :: structVarsToCase
+          | _ => pure ()
+
+    return structVarsToCase.eraseDups
+
+/--
+  Split struct equalities using mk.injEq for all ProvableStruct types
+-/
+def splitProvableStructEq : TacticM Unit := do
+  withMainContext do
+    -- First, find and apply cases on struct variables in equalities
+    let varsToCase ← findStructVarsInEqualities
+
+    if !varsToCase.isEmpty then
+      let mut currentGoal ← getMainGoal
+      for fvarId in varsToCase do
+        try
+          let casesResult ← currentGoal.cases fvarId
+          match casesResult.toList with
+          | [subgoal] =>
+            currentGoal := subgoal.mvarId
+          | _ => continue
+        catch e =>
+          let ldecl ← fvarId.getDecl
+          trace[Meta.Tactic] "Failed to apply cases to struct variable {ldecl.userName}: {e.toMessageData}"
+          continue
+
+      -- Update the goal after cases
+      replaceMainGoal [currentGoal]
+
+    -- Apply mk.injEq lemmas - much simpler approach
+    withMainContext do
+      -- Get environment to check which lemmas exist
+      let env ← getEnv
+      let mut lemmasToApply : List Ident := []
+
+      -- Get all the types that appear in equalities (including inside conjunctions)
+      let ctx ← getLCtx
+      for localDecl in ctx do
+        if localDecl.isImplementationDetail then
+          continue
+
+        let type := localDecl.type
+
+        -- Extract all equalities from this hypothesis (handles conjunctions)
+        let equalities ← extractEqualities type
+
+        for (eqExpr, _, _) in equalities do
+          -- Get the type argument (first argument of Eq)
+          if let some typeExpr := eqExpr.getArg? 0 then
+            -- Get the type constructor for finding mk.injEq
+            let typeExpr' ← whnf typeExpr
+            match typeExpr'.getAppFn with
+            | .const typeName _ =>
+              -- Check if mk.injEq exists for this type
+              let mkInjEqName := typeName ++ `mk ++ `injEq
+              if env.contains mkInjEqName then
+                -- Check if the full type (not just the constructor) has ProvableStruct instance
+                -- For types like MyStruct n (F p), check ProvableStruct (MyStruct n)
+                if ← hasProvableStructInstance typeExpr' then
+                  let lemmaIdent := mkIdent mkInjEqName
+                  if !lemmasToApply.any (fun l => l.getId == mkInjEqName) then
+                    lemmasToApply := lemmaIdent :: lemmasToApply
+            | _ => pure ()
+
+      -- Apply all the lemmas we found
+      for lemmaIdent in lemmasToApply do
+        try
+          evalTactic (← `(tactic| simp only [$lemmaIdent:ident] at *))
```

Comment:
This might do too much when there are multiple conjuncts that mention the same struct type, and only some of them talk about the components. We need first a test case for this.

Reply:
I added a test case `test_with_base_and_non_base`.

The simplification leaves unrelated conjuncts intact. So it's already behaving as expected.